examples/amm/src/contract.rs

```diff
@@ -10,28 +10,41 @@ use amm::{AmmError, ApplicationCall, Message, Operation};
 use async_trait::async_trait;
 use fungible::{Account, FungibleTokenAbi};
 use linera_sdk::{
-    base::{AccountOwner, Amount, ApplicationId, Owner, WithContractAbi},
+    base::{AccountOwner, Amount, ApplicationId, WithContractAbi},
     contract::system_api,
     ensure, ApplicationCallOutcome, Contract, ContractRuntime, ExecutionOutcome, OutgoingMessage,
     Resources, ViewStateStorage,
 };
 use num_bigint::BigUint;
 use num_traits::{cast::FromPrimitive, ToPrimitive};
 
-linera_sdk::contract!(Amm);
+pub struct AmmContract {
+    state: Amm,
+    runtime: ContractRuntime,
+}
+
+linera_sdk::contract!(AmmContract);
 
-impl WithContractAbi for Amm {
+impl WithContractAbi for AmmContract {
     type Abi = amm::AmmAbi;
 }
 
 #[async_trait]
-impl Contract for Amm {
+impl Contract for AmmContract {
     type Error = AmmError;
     type Storage = ViewStateStorage<Self>;
+    type State = Amm;
```

Comment:
Perhaps we could discuss in a separate PR whether the SDK should manage the persistent state at all.